### ISSUE TITLE
Suggestion for a "Restart Option"

### ISSUE THREAD

**Issue by RB61** [JeanLucPons/Kangaroo#10]
It would be nice to save the hash-table into a file every predetermined time lapse  for restarting purposes. It could be useful In case of a crash or maybe if one wants to interrupt the process which may last many days or months in real cases.

**Comment by JeanLucPons** [comment]
Hi,
Yes i will add this 3 options:
-w workfile: Specify file to save work into
 -i workfile: Specify file to load work from
 -wi workInterval: Periodic interval (in seconds) for saving work
as in my BTCCollider project...

**Comment by djarumlights** [comment]
This is great. Had to shut down due to a storm last night and thought about this exact thing.

**Comment by RB61** [comment]
Hi @JeanLucPons  

I figured out the hash-table structure in your program and I was able to write down a function hashTable.WriteHashTable into a binary file and also a hashTable.ReadHashFile from binary file. 
I've just followed the way you defined hashTable.PrintInfo()  which is not used anywhere but was very useful.

Also the exact place where you placed the function hashTable.GetMB() was a good hint.
The writing of the hashtable into a file is the easy part. I juste added the line
       hashTable.WriteHashTable() ; 
in the if statement starting at line 228 in Thread.cpp 
   if(isAlive(params)) {
    }

I think that there is no need for the option  " -wi  workInterval"
because it is already build in when you refresh CPU, GPU Keyrates and other info.

The hard part is to figure out where exactly to insert 
           hashTable.ReadHashTable()   when restarting execution. 

My initial guess is the to change the definition of
bool HashTable::Add(Int *x,Int *d,uint32_t type) {
}
where loading initial hashtable could  be done when the  -i workfile option is used 

Help :)

Je ne suis pas très c++ , suis de l'école FORTRAN lol :)
